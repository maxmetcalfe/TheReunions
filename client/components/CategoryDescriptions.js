import React, { Component } from "react";

class CategoryDescriptions extends Component {

  constructor() {
    super();
    this.baseClass = "coin";
    this.descriptions = {
      five: {
        value: 5,
        text: "All four members present + special occasion"
      },
      four: {
        value: 4,
        text: "All four members present"
      },
      three: {
        value: 3,
        text: "Three members present"
      }, 
      two: {
        value: 2,
        text: "Two members present"
      }
    };
  }

  render() {
    return (
      <table className="category-descriptions">
        <tbody>
          {Object.keys(this.descriptions).map((category) => (
            <tr key={category}>
              <td><div className={this.baseClass + " " + category}>{this.descriptions[category].value}</div></td>
              <td><h3 className={category}>{this.descriptions[category].text}</h3></td>
            </tr>
          ))}
        </tbody>
      </table>
    );
  }
}

export default CategoryDescriptions;
