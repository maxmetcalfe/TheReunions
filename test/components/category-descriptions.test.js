import React from "react"
import renderer from "react-test-renderer"
import CategoryDescriptions from "../../src/components/CategoryDescriptions"

test("CategoryDescriptions", () => {
  let component = renderer.create(<CategoryDescriptions></CategoryDescriptions>)

  expect(component).toMatchSnapshot()
})
