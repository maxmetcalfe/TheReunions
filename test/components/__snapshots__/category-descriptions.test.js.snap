// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryDescriptions 1`] = `
<table
  className="category-descriptions"
>
  <tbody>
    <tr>
      <td>
        <div
          className="coin four"
        >
          4
        </div>
      </td>
      <td>
        <h3
          className="four"
        >
          All four members present.
        </h3>
      </td>
    </tr>
    <tr>
      <td>
        <div
          className="coin three"
        >
          3
        </div>
      </td>
      <td>
        <h3
          className="three"
        >
          Three members present.
        </h3>
      </td>
    </tr>
    <tr>
      <td>
        <div
          className="coin two"
        >
          2
        </div>
      </td>
      <td>
        <h3
          className="two"
        >
          Two members present.
        </h3>
      </td>
    </tr>
  </tbody>
</table>
`;
